r/R/record-batch-reader.R

```diff
@@ -100,7 +100,11 @@ RecordBatchReader <- R6Class("RecordBatchReader",
     read_table = function() Table__from_RecordBatchReader(self),
     Close = function() RecordBatchReader__Close(self),
     export_to_c = function(stream_ptr) ExportRecordBatchReader(self, stream_ptr),
-    ToString = function() self$schema$ToString(),
+    ToString = function() {
+      n_fields_out <- paste0(length(self$schema$fields), " columns", "\n")
+      schema <- self$schema$ToString(truncate = TRUE)
+      paste0(n_fields_out, schema)
+    },
```

Comment:
There is only one other copy of this, so there's not _a huge amount of benefit_ but do you want to factor this into a helper function and use it here + for `Dataset$ToString()`

Reply:
Absolutely, it's good practice so I've done it.